server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -438,6 +439,26 @@ public Map<String, NamedAnalyzer> indexAnalyzers() {
         return Map.of();
     }
 
+    /**
+     * If a mapper natively supports synthetic source, f.e. by constructing it from doc values.
+     * @return true or false
+     */
+    protected boolean supportsSyntheticSourceNatively() {
+        return false;
+    }
+
+    /**
+     * Mappers override this method with native synthetic source support.
+     * If mapper does not support synthetic source, it is generated using generic implementation
+     * in {@link DocumentParser#parseObjectOrField} and {@link ObjectMapper#syntheticFieldLoader()}.
+     *
+     * @return implementation of {@link SourceLoader.SyntheticFieldLoader}
+     */
+    @Override
+    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
+        return SourceLoader.SyntheticFieldLoader.NOTHING;
```

Comment:
This is nothing because `ObjectMapper` already adds all ignored source fields to the source.